### PR TITLE
refactor(storage): remove async_trait for Storage trait

### PR DESCRIPTION
**Pull request by Fedomn**
Part of https://github.com/risinglightdb/risinglight/issues/389

Issue Number: close https://github.com/risinglightdb/risinglight/issues/392

Signed-off-by: Fedomn <fedomn.ma@gmail.com>

### CODE REVIEW

File: src/storage/mod.rs [hunk 1]
```diff
@@ -54,23 +54,33 @@ impl StorageImpl {
...
 
 /// Represents a storage engine.
-#[async_trait]
 pub trait Storage: Sync + Send + 'static {
+    /// the following two result future types to avoid `impl Future` return different types when
+    /// impl `Storage`.
+    type CreateTableResultFuture<'a>: Future<Output = StorageResult<()>> + Sync + Send + 'a
+    where
+        Self: 'a;
+    type DropTableResultFuture<'a>: Future<Output = StorageResult<()>> + Sync + Send + 'a
```
Comment (skyzh) [inline src/storage/mod.rs:63]:
```suggestion
    type DropTableResultFuture<'a>: Future<Output = StorageResult<()>> + Send + 'a
```

Doesn't need to be `Sync`.

File: src/storage/mod.rs [hunk 2]
```diff
@@ -54,23 +54,33 @@ impl StorageImpl {
...
+    fn create_table<'a>(
+        &'a self,
         database_id: DatabaseId,
         schema_id: SchemaId,
-        table_name: &str,
-        column_descs: &[ColumnCatalog],
-    ) -> StorageResult<()>;
+        table_name: &'a str,
+        column_descs: &'a [ColumnCatalog],
+    ) -> Self::CreateTableResultFuture<'_>;
```
Comment (skyzh) [inline src/storage/mod.rs:79]: Better to also use `'a` here?
```suggestion
    ) -> Self::CreateTableResultFuture<'a>;
```

**Review by skyzh — COMMENTED** [review]
Rest LGTM, good work!

File: src/storage/mod.rs (hunk 2 above)
Comment (Fedomn) [reply src/storage/mod.rs:79]: make sense, it's easier to understand

File: src/storage/mod.rs (hunk 1 above)
Comment (Fedomn) [reply src/storage/mod.rs:63]: updated

**Review by skyzh — APPROVED** [review]
LGTM, good work!

**Pull request merged**